### PR TITLE
More tablet manager cleanup.

### PR DESCRIPTION
**Pull request by alainjobart**
@enisoc @sougou 

I am not going to do more until we use better endpoint management and retire serving graph (at which point the health check won't update the tablet, so it won't need to run the 'after an action' workflow.)

### CODE REVIEW

**Comment by alainjobart** [comment]
@aaijazi @michael-berlin Now picking people that are not out of the office for the day...

**Comment by michael-berlin** [comment]
LGTM

Feel free to address my comments. I consider them optional though.

**Comment by alainjobart** [comment]
Thanks! Addressed the comments, will wait for travis to merge.

File: go/vt/tabletmanager/agent_rpc_actions.go [hunk 1]
```diff
@@ -466,7 +466,7 @@ func (agent *ActionAgent) DemoteMaster(ctx context.Context) (myproto.Replication
 	// database. This will in effect close the connection pools to the
 	// database.
 	tablet := agent.Tablet()
-	agent.disallowQueries(tablet.Tablet, "DemoteMaster marks server rdonly")
+	agent.disallowQueries(tablet.Tablet.Type, "DemoteMaster marks server rdonly")
```
Comment (enisoc) [inline go/vt/tabletmanager/agent_rpc_actions.go]: We should be checking the error here, right? We shouldn't continue if we failed to stop in-flight transactions.

**Pull request merged**